DOCS-8098-mma-docsite-v43-duke (#1440)
mulesoft/docs-mule-runtime: modules/ROOT/pages/migration-api-gateways-policies.adoc

Just like before, once the policy template JAR is in Exchange, it will appear in API Manager for APIs that belong
to the same organization where the JAR was uploaded.

[[performing_migration]]

> == Performing a Policy Migration

Reviewer: Performing > Perform (per style guide)

Author: done via suggestion

Author: done via suggestion

> == Performing a Policy Migration

Author: ```suggestion
== Perform a Policy Migration
```

> To perform a policy migration, this example uses the Mule Migration Assistant (MMA), an open source project on GitHub. The procedure requires that you meet the https://github.com/mulesoft/mule-migration-assistant/blob/master/docs/user-docs/migration-tutorial.adoc#prerequisites[Prerequisites] for the the tool. For complete user documentation on MMA, see https://github.com/mulesoft/mule-migration-assistant/blob/master/docs/migration-intro.adoc[Migration to Mule 4 (on GitHub)].

Reviewer: for the the > for the

Author: done via suggestion

Author: done via suggestion

> To perform a policy migration, this example uses the Mule Migration Assistant (MMA), an open source project on GitHub. The procedure requires that you meet the https://github.com/mulesoft/mule-migration-assistant/blob/master/docs/user-docs/migration-tutorial.adoc#prerequisites[Prerequisites] for the the tool. For complete user documentation on MMA, see https://github.com/mulesoft/mule-migration-assistant/blob/master/docs/migration-intro.adoc[Migration to Mule 4 (on GitHub)].

Author: ```suggestion
To perform a policy migration, this example uses the Mule Migration Assistant (MMA), an open source project on GitHub. The procedure requires that you meet the https://github.com/mulesoft/mule-migration-assistant/blob/master/docs/user-docs/migration-tutorial.adoc#prerequisites[Prerequisites] for the tool. For complete user documentation on MMA, see https://github.com/mulesoft/mule-migration-assistant/blob/master/docs/migration-intro.adoc[Migration to Mule 4 (on GitHub)].
```

The MMA command for migrating policies differs from the typical
xref:migration-tool-procedure.adoc#options[command-line options] for migrating
a Mule app in one important way:

* MMA expects the `-projectBasePath` to have both the policy XML and YAML files
in the same directory. The following example assumes that policy files are under
`src/main/policy`.

.Command-line Invocation:
[source,console,linenums]
----
$ java -jar mule-migration-assistant-runner-0.5.1.jar \
 -projectBasePath /Users/me/AnypointStudio/workspace-studio/my-mule3-policy/src/main/policy \
 -muleVersion 4.1.5 \
 -destinationProjectBasePath /Users/me/my-dir/my-migrated-policy
----

Note that the MMA creates the directory for the migrated project through

> the `-destinationProjectBasePath` option. The  `my-migrated-policy` must _not_

Reviewer: The  `my-migrated-policy` > there's an extra space

Author: done via suggestion

Author: done via suggestion

> the `-destinationProjectBasePath` option. The  `my-migrated-policy` must _not_

Author: ```suggestion
the `-destinationProjectBasePath` option. The `my-migrated-policy` must _not_
```

exist before you invoke the command. If you point to a folder that exists

> already, the migration will fail an error like this:

Reviewer: will fail an error > fails with an error...

Author: done via suggestion

Author: done via suggestion

> already, the migration will fail an error like this:

Author: ```suggestion
already, the migration fails with an error like this:
```

`Exception: Destination folder already exists.`

> When the migrator runs successfully, you see a message something like this:

Reviewer: remove "something"

Author: done via suggestion

Author: done via suggestion

> When the migrator runs successfully, you see a message something like this:

Author: ```suggestion
When the migrator runs successfully, you see a message like this:
```
